### ISSUE TITLE
Get the density of Water, imposing the liquid phase, ERROR?

### ISSUE THREAD

**Issue by birdol** [CoolProp/CoolProp#1818]
I got a demo here: http://www.coolprop.org/coolprop/HighLevelAPI.html#imposing-the-phase-optional

```
PropsSI('D','T|liquid',461.1,'P',5e6,'Water')
PropsSI('D','T',597.9,'P|gas',5e6,'Water')
```
Excel show me an error :
```
Input pair variable is invalid and output(s) are non-trivial; cannot do state update : PropsSI("D","T|liquid",461.1,"P",5000000,"Water")
```

**Comment by jowr** [comment]
Are you running the latest version of CoolProp?

**Comment by birdol** [comment]
CoolProp 6.1.0

**Comment by birdol** [comment]
Add phase specification to high-level interface

is this mean, i should download 6.2 version?

**Comment by birdol** [comment]
```
PropsSI('D','T|liquid',461.1,'P',5e6,'Water')
PropsSI('D','T',597.9,'P|gas',5e6,'Water')
```
According to the documentation, 1 would be slow, 2 would be fast, but in fact it's the other way around, 1 would be slower, 2 would be faster.

**Comment by birdol** [comment]
```
    t1 = PropsSI("D", "T", 461.1, "P", 5000000#, "Water")
    t2 = PropsSI("D", "T|liquid", 461.1, "P", 5000000#, "Water")
```
According to the documentation, 1 would be slow, 2 would be fast, but in fact it's the other way around, 1 would be slower, 2 would be faster.

**Comment by birdol** [comment]
```
    t1 = PropsSI("D", "T", 461.1, "P", 5000000#, "Water")
    t2 = PropsSI("D", "T|liquid", 461.1, "P", 5000000#, "Water")
```
According to the documentation, 1 would be slow, 2 would be fast, but in fact it's the other way around, 2 would be slower, 1 would be faster.

**Comment by ibell** [comment]
What is the time for each calc?  Do you get the same result for both?

**Comment by birdol** [comment]
```
Sub coolprop_10000_den()
Dim Time_start As Double
Time_start = Timer
Dim t1 As Double
For i = 0 To 10000 Step 1
    t1 = PropsSI("D", "T", 461.1, "P", 10000 + i / 10000, "Water")
Next
Dim Time_over As Double
Time_over = Timer
MsgBox ("COOLPROP  10000 times den calc cost：" & Format(Time_over - Time_start, "0.0000s"))

End Sub
```
```
Sub coolprop_10000_den_withPH()
Dim Time_start As Double
Time_start = Timer
Dim t1 As Double
For i = 0 To 10000 Step 1
    t1 = PropsSI("D", "T|liquid", 461.1, "P", 10000 + i / 10000, "Water")    
Next
Dim Time_over As Double
Time_over = Timer
MsgBox ("COOLPROP  10000 times den calc cost：" & Format(Time_over - Time_start, "0.0000s"))

End Sub
```

```

**Comment by birdol** [comment]
1 cost about 1.3second， 2 cost about1.5 second 。  time cost is different in different computer. 

2 always going to  take a long time

**Comment by ibell** [comment]
I agree, for PT inputs:
![image](https://user-images.githubusercontent.com/1859947/56043072-c009f180-5cf9-11e9-8e7b-8d476e03e412.png)

I didn't see a similar test for PH inputs, so it's hard to say.

What is happening here is that there is little difference because you are far from the saturation curve, so the full vapor-liquid equilibrium calculation can be skipped (that's the major speedup with imposing the phase).  The phase specification is also *necesssary* if your PT inputs are at or near saturation, to ensure you get the right phase (liquid or gas)

**Comment by birdol** [comment]
ok  i got it 。

another question：“.MIX” works in coolprop？ like refprop？

```
    t1 = PropsSI("D", "T", 273.15 , "P", 2418700, "R410A.MIX")         
    t2 = PropsSI("D", "T", 273.15 , "P", 2418700, "R410A")

```
the second line works much more faseter then first line 。.MIX works?

**Comment by birdol** [comment]
with refprop,  I can use R410A.PPF instead of R410A.MIX，make calc speed up。CoolProp is the same?

**Comment by ibell** [comment]
Yes, and in fact, many calculations are possible with the ppf that would
not be possible with the mix, because the mix is a mixture, and the
routines for mixtures in CoolProp are not complete

**Comment by jowr** [comment]
Is this an issue or can it be tagged "invalid" since it does not cover a bug in CoolProp?

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
